File: CapitalOneInterview/ObjC/TypeaheadViewController.m
//
//  TypeaheadViewController.m
//  CapitalOneInterview
//
//  Created by Dan Deng on 10/23/18.
//  Copyright © 2018 Dan Deng. All rights reserved.
//

#import "TypeaheadViewController.h"
#import "SearchResultCell.h"
#import "SearchResult.h"

@interface TypeaheadViewController ()

@end

@implementation TypeaheadViewController

// MARK: - Properties

@synthesize searchField;
@synthesize tableView;

// MARK: - Lifecycle

- (instancetype)init {
    self.searchField = [[UITextField alloc] init];
    self.tableView = [[UITableView alloc] init];
    return [super init];
}

- (void)viewDidLoad {
    [super viewDidLoad];

    // Setup searchField
    [self.view addSubview:self.searchField];
    [self.searchField addTarget:self action:@selector(searchTextDidUpdate:) forControlEvents:UIControlEventEditingChanged];

    // Setup tableView
    [self.view addSubview:self.tableView];
    [self.tableView setDataSource:self];
    [self.tableView setDelegate:self];
    [self.tableView registerClass:[SearchResultCell class] forCellReuseIdentifier:SearchResultCell.reuseID];
}

// MARK: - Text field handler

// Called when the text in searchField is updated
- (void)searchTextDidUpdate:(UITextField*)textField {
    // TODO: Candidate implements this
}

// MARK: - Table view data source

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    // TODO: Candidate implements this
    return 0;
}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    // TODO: Candidate implements this
    return [[UITableViewCell alloc] init];
}

@end

File: CapitalOneInterview/Swift/TypeaheadViewController.swift
//
//  TypeaheadViewController.swift
//  CapitalOneInterview
//
//  Created by Dan Deng on 9/5/18.
//  Copyright © 2018 Dan Deng. All rights reserved.
//

import UIKit

class TypeaheadViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {

    // MARK: - Properties

    let searchField = UITextField()

    let tableView = UITableView()

    let autocompleteService = AutocompleteService()

    // MARK: - Lifecycle

    override func viewDidLoad() {
        super.viewDidLoad()
        
        // Setup searchField
        self.view.addSubview(self.searchField)
        self.searchField.addTarget(self, action: #selector(searchTextDidUpdate), for: .editingChanged)

        // Setup tableView
        self.view.addSubview(self.tableView)
        self.tableView.register(SearchResultCell.self, forCellReuseIdentifier: SearchResultCell.reuseIdentifier)
        self.tableView.delegate = self
        self.tableView.dataSource = self
    }

    // MARK: - Table view data source

    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        // TODO: Candidate implements this
        return 0
    }

    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        // TODO: Candidate implements this
        return UITableViewCell()
    }

    // MARK: - Text field handler

    // Called when the text in searchField is updated
    @objc func searchTextDidUpdate(textField: UITextField) {
        // TODO: Candidate implements this
    }

}
